### ISSUE TITLE
Declaration File missing

### ISSUE THREAD

**Issue by jpmaligne** [danalloway/react-country-flag#16]
What I have done : `npm install --save react-country-flag`

react-country-flag folder appears well in my node-modules.

Then : `import ReactCountryFlag from "react-country-flag";` from a .ts file

Gets me this error :

```
Could not find a declaration file for module 'react-country-flag'. '/home/{...}/node_modules/react-country-flag/dist/index.js' implicitly has an 'any' type.
  Try `npm install @types/react-country-flag` if it exists or add a new declaration (.d.ts) file containing `declare module 'react-country-flag';`  TS7016
```

**Comment by danalloway** [comment]
that is correct, I do not have any type declarations for this project

I would recommended following their suggestion and adding a local type declaration for this library in your project

**Comment by webhacking** [comment]
```
declare module 'react-country-flag' {
  import * as React from 'react';
  export interface ReactCountryFlagProps {
    code: string;
  }
  export default class ReactCountryFlag extends React.Component<ReactCountryFlagProps, any> {}
}
`
``

**Comment by thibaultboursier** [comment]
@danalloway @webhacking Is there something new about declaration file? I can create PR if you want.

**Comment by danalloway** [comment]
@thibaultboursier want to wait for v2 to land, then create a PR for that one?

**Comment by thibaultboursier** [comment]
@danalloway OK!

**Comment by danalloway** [comment]
v2 is out, ready for that declaration file if you're still up for it

**Comment by thibaultboursier** [comment]
Ok, i'll doit that soon.

**Comment by danalloway** [comment]
@thibaultboursier just checking in on this, if you're still going to do it i'll leave this issue open, otherwise please close it out for me, thank you

**Comment by thibaultboursier** [comment]
@danalloway Yes, still going to do it.

**Comment by DATeclipse** [comment]
@thibaultboursier @danalloway - Just wanted to ping to see if this is still happening?

**Comment by FabianoLothor** [comment]
> @thibaultboursier @danalloway - Just wanted to ping to see if this is still happening?

Yes, I still getting this error, where is the `.ts` file?

**Comment by danalloway** [comment]
@DATeclipse @FabianoLothor waiting on a PR for this, feel free to take a stab at it I don't know if anyone else is at the moment

**Comment by FabianoLothor** [comment]
@danalloway I fix it in my project with this workaround:

```
# New file /src/external.d.ts
declare module "react-country-flag";
```

**Comment by r3wt** [comment]
Declarations for v2:

```ts
declare module 'react-country-flag' {
  import * as React from 'react';
  export interface ReactCountryFlagProps extends React.DetailedHTMLProps<React.LabelHTMLAttributes<HTMLImageElement>,HTMLImageElement> {
    cdnSuffix?:string;
    cdnUrl?:string;
    countryCode: string;
    svg?: boolean;
    style?: React.CSSProperties;
  }
  const ReactCountryFlag:React.FC<ReactCountryFlagProps> = ({
    cdnSuffix='svg',
    cdnUrl='https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.4.3/flags/4x3/',
    svg=false,
    style={},
    countryCode
  })=>React.ReactNode;// return should be HTMLImageElement or HTMLSpanElement but close enough for me.
  export default ReactCountryFlag;
};
```

Now, this is technically not complete, because really the default export should be overloaded to show the effect of the `svg` prop on the return type as noted in the comment, but i mean, most people probably won't care much about that, so just returning React.ReactNode should be good enough, imo. @here feel free to expand on this.

**Comment by r3wt** [comment]
Here are some declarations i wrote for v2. It's not perfect, but it seems to work for me. Maybe someone who is better at TS can come along and make it perfect.

```ts
declare module 'react-country-flag' {
  import * as React from 'react';
  export interface ReactCountryFlagProps<T> extends React.DetailedHTMLProps<React.LabelHTMLAttributes<T>,T> {
    cdnSuffix?:string;
    cdnUrl?:string;
    countryCode: string;
    svg?: boolean;
    style?: React.CSSProperties;
  }
  const ReactCountryFlag:React.FC<ReactCountryFlagProps<HTMLDivElement|HTMLImageElement>> = ({
    cdnSuffix='svg',
    cdnUrl='https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.4.3/flags/4x3/',
    svg=false,
    style={},
    countryCode
  })=>React.ReactNode;// return should be HTMLImageElement or HTMLSpanElement but close enough for me.
  export default ReactCountryFlag;
};
```

**Comment by thibaultboursier** [comment]
Hello @danalloway @FabianoLothor @r3wt,

I opened a PR to add TypeScript and generate declaration files each time project is built.
You can check it [here](https://github.com/danalloway/react-country-flag/pull/31).

**Comment by r3wt** [comment]
@thibaultboursier nice work! the only thing is, i'm not sure how the repo owner will feel about it, as it might make it difficult to maintain if they don't know typescript.

**Comment by thibaultboursier** [comment]
@r3wt Thanks. Let's discuss with @danalloway about that :-) Anyway, with a declaration file manually written, we would also have to maintain it with TypeScript.

**Comment by danalloway** [comment]
@r3wt @thibaultboursier I may prefer the manual / static file
this code base won't change much as it's a pretty dead simple library
will update you all ASAP when I make the change

**Comment by alielkhateeb** [comment]
Any update on this?

**Comment by danalloway** [comment]
working on a Typescript version of this library, should publish today

**Comment by danalloway** [comment]
just released version v3.0.1 with Typescript support
